src/storage/txn/sched_pool.rs

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::cell::RefCell;
+use std::time::Duration;
+
+use prometheus::local::*;
+use tikv_util::collections::HashMap;
+use tikv_util::future_pool::Builder as FuturePoolBuilder;
+use tikv_util::future_pool::FuturePool;
+
+use crate::storage::metrics::*;
+use crate::storage::{Engine, Statistics, StatisticsSummary};
+
+pub struct SchedLocalMetrics {
+    stats: HashMap<&'static str, StatisticsSummary>,
+    processing_read_duration: LocalHistogramVec,
+    processing_write_duration: LocalHistogramVec,
+    command_keyread_histogram_vec: LocalHistogramVec,
+}
+
+thread_local! {
+     static TLS_SCHED_METRICS: RefCell<SchedLocalMetrics> = RefCell::new(
+        SchedLocalMetrics {
+            stats: HashMap::default(),
+            processing_read_duration: SCHED_PROCESSING_READ_HISTOGRAM_VEC.local(),
+            processing_write_duration: SCHED_PROCESSING_WRITE_HISTOGRAM_VEC.local(),
+            command_keyread_histogram_vec: KV_COMMAND_KEYREAD_HISTOGRAM_VEC.local(),
+        }
+    );
+}
+
+#[derive(Clone)]
+pub struct SchedPool<E: Engine> {
+    pub engine: E,
+    pub pool: FuturePool,
+}
+
+impl<E: Engine> SchedPool<E> {
+    pub fn new(engine: E, pool_size: usize, name_prefix: &str) -> Self {
+        let pool = FuturePoolBuilder::new()
+            .pool_size(pool_size)
+            .name_prefix(name_prefix)
+            .on_tick(move || tls_flush())
+            .before_stop(move || tls_flush())
+            .build();
+        SchedPool { engine, pool }
+    }
+}
+
+pub fn build_sched_pool(pool_size: usize, name_prefix: &str) -> FuturePool {
```

Comment:
The name is quite confusing.. `build_sched_pool` does not build a `SchedPool` but a `FuturePool` instead.